### PR TITLE
👔  Implement creator and increase leak coverage

### PR DESCRIPTION
**Pull request by samccone**
We now have memory tests across the board for every element :)

---------

**MaterialButton** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
0|0 bytes|0

:fire: :fire: **MaterialSpinner** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
145|0 bytes|0

**MaterialTooltip** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
0|0 bytes|0

**MaterialCheckbox** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
0|0 bytes|0

**MaterialIconToggle** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
0|0 bytes|0

:fire: **MaterialDataTable** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
5|0 bytes|0

:fire: :fire: :fire: **MaterialLayout** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
260|0 bytes|35

**MaterialMenu** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
0|0 bytes|0

**MaterialProgress** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
0|0 bytes|0

:fire: :fire:  **MaterialRadio** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
30|0 bytes|20

**MaterialRipple** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
0|0 bytes|0

**MaterialSlider** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
0|0 bytes|0

:fire: **MaterialSnackbar** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
5|0 bytes|0

**MaterialSwitch** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
0|0 bytes|0

**MaterialTabs** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
5|0 bytes|0

**MaterialTextfield** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
0|0 bytes|0

### CODE REVIEW

File: test/unit/creator.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
+module.exports =  {
```
Comment (surma) [inline test/unit/creator.js]: Question: For consistencies sake; Do we just wanna move to inline html strings? I think it is more readable and that benefit definitely outweighs performance in tests, imo.

Comment (samccone) [inline test/unit/creator.js]: yeah maybe, I just took what was in the tests and moved it in here :) but now we can refactor away :D

**Comment by samccone** [comment]
-- not sure what is up with the style checker right now... seems unrelated kinda --

File: test/unit/creator.js (hunk 1 above)
Comment (surma) [inline test/unit/creator.js]: k

**Comment by surma** [comment]
For documentation purposes: @samccone realized that our mocha tests are actually running inside PhantomJS, which is why `require()` is not available. Needs browserify or something.

**Comment by webmaven** [comment]
What needs to be done to move this forward?

**Comment by surma** [comment]
Either rewrite the tests to work without a DOM so they can run in Node rather than PhantomJS or use browserify (or similar) to make @samccone’s work run in the browser.

**Comment by samccone** [comment]
I can close this, or... I mean eventually hit this :)

**Comment by surma** [comment]
@samccone Your choice, really. I guess having a proper testing code base with `require()` and everything is not a bad thing for v2, so no harm in leaving it open to remind us.

**Comment by samccone** [comment]
yeah, it just comes down to :clock1: to do it

**Comment by webmaven** [comment]
This issue should probably be marked as 'Milestone v2', then.

**Comment by sgomes** [comment]
Hello there! We're currently doing some housecleaning on the repository to prepare for the big changes to v2, and part of this is to close the mdl-1.1 and mdl-1.2 branches, and move all 1.1 and 1.2 PRs to the mdl-1.x branch. Since it's not possible to target PRs at a new branch, we're closing this PR and asking you to reopen against mdl-1.x if you think it's still valid.

Thank you!

**Pull request closed without merging**